currency/code.go

```diff
@@ -275,21 +282,27 @@ func (c Code) String() string {
 	if c.Item == nil {
 		return ""
 	}
-	if c.UpperCase {
+	if !c.Item.CaseSensitive || c.upperCase {
 		return c.Item.Symbol
 	}
 	return c.Item.Lower
 }
 
 // Lower converts the code to lowercase formatting
 func (c Code) Lower() Code {
-	c.UpperCase = false
+	if c.Item != nil && !c.Item.CaseSensitive {
+		return c
+	}
+	c.upperCase = false
 	return c
 }
 
 // Upper converts the code to uppercase formatting
 func (c Code) Upper() Code {
-	c.UpperCase = true
+	if c.Item != nil && !c.Item.CaseSensitive {
+		return c
+	}
+	c.upperCase = true
 	return c
```

Comment:
```suggestion
// Upper flags the Code to use UpperCase formatting, but does not change Symbol
// If Code cannot be uppercased then it will return Code undchanged
func (c Code) Upper() Code {
	if c.Item == nil || c.Item.Symbol != c.Item.Lower {
		// Symbol is uppercased already, so if they're equal, then there's no uppercase representation
		return c
	}
	c.UpperCase = true
	return c
}
```

Reply:
Hopefully this resolves the issue; [50191fd](https://github.com/thrasher-corp/gocryptotrader/pull/1593/commits/50191fdd58d5c0ca2cb4129fde7077be475dfd88) Thanks

Reply:
Suspect you meant 1aa4da4 🙂

Reply:
Yes, Seems I have done a boo boo with respect to bitfinex tests.